### DISCUSSION TITLE
Idea: Respect generated files, as marked in gitattributes

### DISCUSSION THREAD

**Discussion by bmulholland** [crate-ci/typos#1111] [Ideas]
Github (and maybe other things?) use a flag in .gitattributes to mark files as being generated. Most people probably don't want generated files checked for typos. If typos-cli could read that file and ignore files marked as generated, it would centralize the list of generated files.

https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github

**Comment by epage** [comment]
There is also an in-source marker `@generated` that is generally supported by different tools.  We'd need to decide if we supported that also.

Currently, we have no special git handling, so we'd need to be careful in defining the semantics of how this would interact with gitattributes.

We'd also likely want to put this behind a config to be overriden.  We might want to keep it off by default to maintain existing behavior.
